processor/integrations/integrations.go

```diff
@@ -19,16 +20,31 @@ var (
 	destTransformURL, userTransformURL string
 	customDestination                  []string
 	whSchemaVersion                    string
+	postParamentersTFields               []string
 )
 
 func init() {
 	loadConfig()
+	populatePostParamenterFields()
 }
 
 func loadConfig() {
 	destTransformURL = config.GetEnv("DEST_TRANSFORM_URL", "http://localhost:9090")
 }
 
+// This is called in init and it should be a one time call. Having reflect calls happening during runtime is not a great idea.
+// We unmarshal json response from transformer into PostParametersT struct.
+// Since unmarshal doesn't check if the fields are present in the json or not and instead just initialze to zero value, we have to manually do this check on all fields before unmarshaling
+// This function gets a list of fields tagged as json from the struct and populates in postParamentersTFields
+
+func populatePostParamenterFields() {
+	v := reflect.TypeOf(PostParametersT{})
+	postParamentersTFields = make([]string, v.NumField(), v.NumField())
```

Comment:
https://www.calhoun.io/how-to-use-slice-capacity-and-length-in-go/

This is a good article to read. We need to clean up our overall code. But I think we should follow one of these standards.

```
	postParamentersTFields = make([]string, v.NumField())
    for i := 0; i < v.NumField(); i++ {
		postParamentersTFields[i] = strings.Split(v.Field(i).Tag.Get("json"), ",")[0]
    }
``` 

Or 

```
	postParamentersTFields = make([]string, 0, v.NumField())
    for i := 0; i < v.NumField(); i++ {
		postParamentersTFields = append(postParamentersTFields, strings.Split(v.Field(i).Tag.Get("json"), ",")[0])
    }
```

Reply:
Actually giving length and capacity both same value is redundant and instead just giving length should suffice. 

Alternatively since I know the length prematurely, I could have just created an array instead of a slice.

Reply:
I prefer the second approach using append. I feel it reads better as it will be clear that we are making up the list (appending) as we go.